### PR TITLE
Added a bandaid for the panics in knn-search algorithm.

### PR DESCRIPTION
**Pull request by nishaq503**
I will revisit the algorithms later for better fixes.

### CODE REVIEW

File: abd-clam/src/cakes/knn/greedy_sieve.rs
Comment (morganprior) [inline abd-clam/src/cakes/knn/greedy_sieve.rs:1]: We don't need to justify unreachables anymore?

File: abd-clam/src/cakes/knn/greedy_sieve.rs [hunk 1]
```diff
@@ -37,19 +37,13 @@ where
...
     while hits.len() < k
         || (!candidates.is_empty()
-            && (hits.peek().unwrap_or_else(|| unreachable!("`hits` is non-empty")).1 .0
-                >= candidates
-                    .peek()
-                    .unwrap_or_else(|| unreachable!("`candidates` is non-empty"))
-                    .1
-                     .0))
+            && hits.peek().map_or_else(|| unreachable!(), |(_, &OrdNumber(d))| d)
+                >= candidates.peek().map_or_else(|| unreachable!(), |(_, &RevNumber(d))| d))
```
Comment (morganprior) [inline abd-clam/src/cakes/knn/greedy_sieve.rs:41]: Should you justify unreachables here?

**Review by morganprior — APPROVED** [review]
Do you need to version bump? Also, if you still want unreachables justified, you should add them back in where I commented. Other than that, looks good to merge!

File: abd-clam/src/cakes/knn/greedy_sieve.rs (hunk 1 above)
Comment (nishaq503) [reply abd-clam/src/cakes/knn/greedy_sieve.rs]: Done!

**Review by morganprior — APPROVED** [review]
LGTM!

**Pull request merged**